From 177603b21a4a99b95e924a78e932a23759490bc8 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 24 Jul 2024 11:31:42 +0200
Subject: [PATCH 1043/1695] clk: stm32mp25: do not access BSEC if
 CFG_STM32_BSEC3 isn't set

BSEC is accessible either by the TDCID, or by the BootROM after a reset.
When the CFG_STM32_BSEC3 switch isn't embedded (i.e. when M33 is TDCID),
do not call BSEC functions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 688835cf8e1b603feffa58e957c339e20a98d71f clk: stm32mp25: do not access BSEC if CFG_STM32_BSEC3 isn't set]
Change-Id: Ic38f6450109cc9f463bc1b9b365963e65f0ff466
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/395777
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index a84f52d2c..49ed6a391 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -3089,12 +3089,16 @@ static const struct clk_ops clk_stm32_rif_gate_ops = {
 	.restore_context = clk_stm32_rif_gate_pm_restore,
 };
 
-static TEE_Result clk_stm32_rif_gate_dbg_enable(struct clk *clk)
+static TEE_Result clk_stm32_rif_gate_dbg_enable(struct clk *clk __maybe_unused)
 {
+#ifdef CFG_STM32_BSEC3
 	if (!stm32_bsec_self_hosted_debug_is_enabled())
 		return TEE_ERROR_ACCESS_DENIED;
 
 	return clk_stm32_rif_gate_enable(clk);
+#else
+	return TEE_SUCCESS;
+#endif
 }
 
 static const struct clk_ops  clk_stm32_rif_gate_dbg_ops = {
-- 
2.47.2

